packages/dataviews/src/filter-and-sort-data-view.ts

```diff
@@ -32,14 +32,14 @@ const EMPTY_ARRAY: Data = [];
  *
  * @return Filtered, sorted and paginated data.
  */
-export function filterSortAndPaginate(
-	data: Data,
+export function filterSortAndPaginate< T extends Item >(
+	data: T[],
 	view: View,
-	fields: Field[]
+	fields: Field< T >[]
 ): { data: Data; paginationInfo: { totalItems: number; totalPages: number } } {
 	if ( ! data ) {
 		return {
-			data: EMPTY_ARRAY,
+			data: EMPTY_ARRAY as T[],
```

Comment:
This `as` is required because `EMPTY_ARRAY` cannot be typed generically, though is used to provide a stable object reference for an empty data array.

Reply:
What's the point of this? I'm assuming it's because `EMPTY_ARRAY` is stable?

I think we can change its type to `[]` and that will satisfy the return type without requiring this assertion.

Reply:
I haven't used that before, I'll try it.

Reply:
That worked. Thanks @sirreal, I will be using that later.